tools/cabana/chartswidget.cc

```diff
@@ -469,6 +469,16 @@ void ChartView::updatePlot(double cur, double min, double max) {
     axis_x->setRange(min, max);
     updateAxisY();
   }
+
+  // Show points when zoomed in enough
+  for (auto &s : sigs) {
+    auto begin = std::lower_bound(s.vals.begin(), s.vals.end(), axis_x->min(), [](auto &p, double x) { return p.x() < x; });
+    auto end = std::lower_bound(s.vals.begin(), s.vals.end(), axis_x->max(), [](auto &p, double x) { return p.x() < x; });
+
+    int pixels_per_point = width() / (end - begin);
```

Comment:
Do we need to make sure not to divide by 0 ?

Reply:
Looks like `pixels_per_point` will just be 0 when end==begin, but will add a check for completeness.